### DISCUSSION TITLE
gmysql/BIND backend not handling wildcards properly

### DISCUSSION THREAD

**Discussion by brainyde** [PowerDNS/pdns#13214] [Q&A] [answered]
After moving zones from tinydns-backend to gmysql/BIND-backend we notice changes in handling of wildcard-entries. We think that the handling in the gmysql and BIND-backend is wrong.

Zone (shortened to relevant entries):
*.example.com       600     IN      A       127.0.0.1
*.staging.example.com       600     IN      A       127.0.0.2

dig @127.0.0.1 sdjsdflksdfkl.example.com +short
127.0.0.1
dig @127.0.0.1 sdjsdflksdfkl.staging.example.com +short
127.0.0.2
dig @127.0.0.1 staging.example.com +short
<empty/nothing>

The entry "staging.example.com" should be covered by the wildcard-entry of "*.example.com" but is not.

PowerDNS-version is 4.7.3-2 (Debian Package)

**Comment by Habbie** [comment]
This is correct behaviour. I'm moving this to Discussions.

**Reply by Habbie** [reply]
https://www.rfc-editor.org/rfc/rfc4592#section-2.2.2 is a very short explanation of why this is the right (but often annoying!) behaviour

**Reply by Habbie** [reply, marked as answer]
http://cr.yp.to/djbdns/axfr-get.html notes

>  axfr-get does not precisely simulate BIND's handling of *.dom. Under BIND, records for *.dom do not apply to y.dom or anything.y.dom if there is a normal record for x.y.dom. With axfr-get and tinydns, the records apply to y.dom and anything.y.dom except x.y.dom. 

The same "bug" is in `tinydns-data`, which means our tinydnsbackend cannot do the right thing either.

**Reply by brainyde** [reply]
With tinydns-backend (we have version 4.4.3) its working


tinydns-data:
```
+*.example.com:127.0.0.1:600
+*.staging.example.com:127.0.0.2:600
```

```
dig @127.0.0.1 example.com A +short
<empty>
dig @127.0.0.1 test.example.com A +short
127.0.0.1
dig @127.0.0.1 test.staging.example.com A +short
127.0.0.2
dig @127.0.0.1 staging.example.com A +short
127.0.0.1
```

**Reply by Habbie** [reply]
yes - this is the *wrong* behaviour, but we cannot do the right thing here, because `data.cdb` does not contain the `staging` Empty Non Terminal
